packages/transactions/src/OutboundShipment/ListView/ListView.tsx

```diff
@@ -101,81 +98,23 @@ export const OutboundShipmentListViewComponent: FC = () => {
     pagination,
   } = useListData({ key: 'color' }, 'transaction', OutboundShipmentListViewApi);
 
-  const columns = useColumns<Transaction>([
-    {
-      ...getNameAndColorColumn<Transaction>((row, color) => {
-        onUpdate({ ...row, color: color.hex });
-      }),
-
-      key: 'name',
-      label: 'label.name',
-      sortable: true,
-      width: 150,
-      minWidth: 200,
-      maxWidth: 250,
-      align: 'left',
-    },
-    {
-      label: 'label.type',
-      key: 'type',
-      width: 150,
-      minWidth: 150,
-      maxWidth: 100,
-      align: 'left',
-    },
-    {
-      label: 'label.status',
-      key: 'status',
-      width: 100,
-      minWidth: 100,
-      maxWidth: 100,
-      align: 'left',
-    },
-    {
-      label: 'label.entered',
-      key: 'entered',
-      format: ColumnFormat.date,
-      width: 100,
-      minWidth: 100,
-      maxWidth: 100,
-      align: 'left',
-    },
-    {
-      label: 'label.confirmed',
-      key: 'confirmed',
-      format: ColumnFormat.date,
-      width: 100,
-      minWidth: 100,
-      maxWidth: 100,
-      align: 'left',
-    },
-
-    {
-      label: 'label.invoice-number',
-      key: 'invoiceNumber',
-      width: 75,
-      minWidth: 75,
-      maxWidth: 75,
-      align: 'left',
-    },
-    {
-      label: 'label.total',
-      key: 'total',
-      width: 75,
-      minWidth: 75,
-      maxWidth: 75,
-      align: 'right',
-    },
-    {
-      label: 'label.comment',
-      key: 'comment',
-      width: 150,
-      minWidth: 150,
-      maxWidth: 450,
-      align: 'left',
-    },
-    getCheckboxSelectionColumn() as ColumnDefinition<Transaction>,
-  ]);
+  const onColorUpdate = (row: Transaction, color: Color) => {
+    onUpdate({ ...row, color: color.hex });
+  };
+
+  const columns = useColumns(
+    new ColumnSetBuilder<Transaction>()
+      .addColumn(getNameAndColorColumn(onColorUpdate))
+      .addColumn('type')
+      .addColumn('status')
+      .addColumn('invoiceNumber')
+      .addColumn('confirmed')
+      .addColumn('entered')
+      .addColumn('total')
+      .addColumn('comment')
+      .addColumn('selection')
+      .build()
+  );
```

Comment:
Again, passing `['type', 'status', 'invoiceNumber', 'confirmed', 'entered', 'total', 'comment', 'selection']` into `useColumns` (And, maybe read this array from `localStorage`, if set) then having `useColumns` handle the builder would be better.

Reply:
for sure - that would be great to do; both the passing of an array ( though the `addColumn` syntax is nice and clear anyway, probably more readable in fact ) and the support for column user preferences